File: cmps101/pa2/Lex.c
//Alix Feinsod
//afeinsod
//CMPS 101 PA2
//Lex.c - program to alphabetize files

#include<stdio.h>
#include<stdlib.h>
#include<string.h>
#include"List.h"

#define MAX_LEN 160

int main(int argc, char * argv[]){

   	FILE *in, *out;
   	char line[MAX_LEN];

   //checks for correct usage
        if( argc != 3 ){
      		printf("Usage: Lex <input file> <output file>\n");
      		exit(1);
   	} 

       	else {
	// Open files for reading and writing
            in = fopen(argv[1], "r");
	    out = fopen(argv[2], "w");
   	    if( in==NULL ){
      		printf("Unable to open file %s for reading\n", argv[1]);
      		exit(1);
   	    }
   	    if( out==NULL ){
      		printf("Unable to open file %s for writing\n", argv[2]);
      		exit(1);
    	    }

	// reads file to get number of lines
            int lines = 0;
            while( fgets(line, MAX_LEN, in) != NULL) {
                lines++;
            }
            fclose(in);
	
	//creates an array of Strings
	    char **S = malloc(lines * sizeof(char *));
	    lines = 0;

	//reopens the file to read in the lines
            in = fopen(argv[1], "r");

        //Adds all lines of input file to list of strings.
            while( fgets(line, MAX_LEN, in) != NULL) {
	      S[lines] = malloc(sizeof(line));
	      strcpy(S[lines], line);
	      lines++;
   	    }

	//sorting into indexList
            List L = newList();
	    append(L, 0);
            for (int i = 1; i < lines; i ++) {
	    
		moveTo(L, 0);
		while ((getIndex(L) >= 0) && (strcmp(S[getElement(L)],S[i])<0)) {
		        
			moveNext(L);
		}
		if (getIndex(L) < 0) {
			append(L, i);
		}
		else {
		        insertBefore(L,i);
		}
	    }

        //prints lines to doc
	    fprintf(out, "%s \n", S[front(L)]);
	    free(S[front(L)]);
            for(moveTo(L, 1); getIndex(L)>=0; moveNext(L)){
	      fprintf(out, "%s", S[getElement(L)]);
	      free(S[getElement(L)]);
	    }
	//closes input and output files
            fclose(in);
            fclose(out);
	//frees everything

	    free(S);
	    freeList(&L);
	    return(0);
	}
}

File: cmps101/pa1/Lex.java
//Alix Feinsod
//afeinsod
//CMPS 101 PA1
//Lex.java - program to alphabetize files

import java.io.*;
import java.util.Scanner;

public class Lex {

    public static void main(String[] args) throws IOException {
        if (args.length != 2) {
            System.err.println("Usage: Lex infile outfile");
            System.exit(1);
        } else {
            BufferedReader reader = new BufferedReader(new FileReader(args[0]));
            int lines = 0;
            while (reader.readLine() != null) {
                lines++;
            }
            reader.close();
            String[] S = new String[lines];
            lines = 0;
            Scanner in = null;
            PrintWriter out = null;
            String line = null;
            in = new Scanner(new File(args[0]));
            out = new PrintWriter(new FileWriter(args[1]));

            //Adds all lines of input file to list of strings.
            while (in.hasNextLine()) {
                line = in.nextLine();
                S[lines++] = line;
            }
            in.close();
            
	    //sorting into indexList
            List L = new List();
	    L.append(0);
            for (int i = 1; i < S.length; i ++) {
		L.moveTo(0);
		while ((L.getIndex() >= 0) && (S[L.getElement()].compareTo(S[i])<0)) {
			L.moveNext();
		}
		if (L.getIndex() < 0) {
			L.append(i);
		}
		else {
			L.insertBefore(i);
		}
	    }

            //prints lines to doc
            for(L.moveTo(0); L.getIndex()>=0; L.moveNext()){
         	out.println(S[L.getElement()]);
	    }
            out.close();
        }
    }
}